frontend/public/actions/features.ts

```diff
@@ -49,6 +50,16 @@ const detectOpenShift = dispatch => coFetchJSON(openshiftPath)
       : handleError(err, FLAGS.OPENSHIFT, dispatch, detectOpenShift)
   );
 
+const detectBaremetalPlatform = dispatch =>
+  k8sGet(InfrastructureModel, 'cluster')
+    .then((infra: K8sResourceKind) =>
+      getInfrastructurePlatform(infra) === 'BareMetal' && dispatch(setFlag(FLAGS.BAREMETAL, true)))
+    .catch((err) => {
```

Comment:
It seems that existing `detectFoo` functions pass success and error handlers as part of `then` invocation.

Suggested change:

```ts
const detectBaremetalPlatform = dispatch => k8sGet(InfrastructureModel, 'cluster')
  .then(
    (infra: K8sResourceKind) => {
      dispatch(setFlag(FLAGS.BAREMETAL, getInfrastructurePlatform(infra) === 'BareMetal'))
    },
    (err) => {
      _.get(err, 'response.status') === 404
        ? dispatch(setFlag(FLAGS.BAREMETAL, false))
        : handleError(err, FLAGS.BAREMETAL, dispatch, detectBaremetalPlatform);
    }
  );
```

Reply:
I believe that handling errors using `onRejected` instead of using `catch` is often considered anti-pattern as using `catch` will also catch the errors occurring during `onFulfilled`.

I've aligned the code to other 'detect' functions though.